### PR TITLE
Switch to bull queue

### PR DESCRIPTION
**Pull request by Emmanuel-Develops**

### CODE REVIEW

File: api/app/utils/cron.ts [hunk 1]
```diff
@@ -1,61 +1,110 @@
...
-import { getUnixTimeFromHours } from "../utils/review.inference";
+import { TranscriptStatus } from "../types/transcript";
+import { buildIsExpiredAndNotArchivedCondition, getUnixTimeFromHours } from "../utils/review.inference";
 import { EXPIRYTIMEINHOURS } from "./constants";
 import { deleteCache } from "../db/helpers/redis";
-import { redis } from "../db";
+import { REDIS_HOST, REDIS_PASSWORD, REDIS_PORT, redis } from "../db";
 
-const CronJob = cron.CronJob;
+const resetRedisCachedPages = async () => {
```
Comment (Extheoisah) [inline api/app/utils/cron.ts:11]: Can you put this in the redis helper file? We're reseting the transcript pages in some other places in the transcript controller also

File: api/package.json [hunk 2]
```diff
@@ -33,6 +33,7 @@
   "dependencies": {
     "@node-lightning/invoice": "^0.28.0",
     "axios": "^1.4.0",
+    "bull": "^4.10.4",
     "cors": "^2.8.5",
     "cron": "^2.3.0",
```
Comment (Extheoisah) [inline api/package.json:38]: We can remove cron as a dependency if we no longer use it

File: api/app/utils/cron.ts [hunk 3]
```diff
@@ -1,61 +1,110 @@
...
+const dailyCheckForMissedExpiredReviews = new Queue("daily-cron-expired", {
+  redis: {
+    port: REDIS_PORT,
+    host: REDIS_HOST,
+    password: REDIS_PASSWORD,
+  },
+  defaultJobOptions: { repeat: { cron: "0 0 * * *" } },
+})
+
+export const startDailyExpiredReviewsCheck = async () => await dailyCheckForMissedExpiredReviews.add({})
```
Comment (Extheoisah) [inline api/app/utils/cron.ts:76]: Where would we call this? I can't find where you're using this function

Comment (Emmanuel-Develops) [reply api/app/utils/cron.ts:76]: In the same file at the end, or in `server.ts` after sequelize sync. Bull is smart enough not to add the same instance of a repeatable job to a queue.
So I didn't call this on purpose for a reason. I'll drop it on discord

File: api/package.json (hunk 2 above)
Comment (Emmanuel-Develops) [reply api/package.json:38]: alright

File: api/app/utils/cron.ts (hunk 1 above)
Comment (Emmanuel-Develops) [reply api/app/utils/cron.ts:11]: sure

File: api/app/utils/cron.ts (hunk 3 above)
Comment (Extheoisah) [reply api/app/utils/cron.ts:67]: Okay. But why aren't you calling the function?

Comment (Emmanuel-Develops) [reply api/app/utils/cron.ts:67]: its on discord boss 👀
TLDR; small chance we clear `claimedby` and `queuestatus` of a transcript that got claimed after it was requeued using the previous implementation. Then again we don't know when the previous implementation stopped working, so it's more or less a non issue

I'm calling it now tho

**Review by Extheoisah — APPROVED** [review]
